tests/beaker_tests/cisco_pim/test_cisco_pim.rb

```diff
@@ -0,0 +1,238 @@
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#   Can be used with :pim for mixed title/pim testing. If mixing, :pim values will
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+#
+tests['title_patterns'] = {
+  manifest_props: "
+    ssm_range   => '224.0.0.0/8 225.0.0.0/8',
+  ",
+  resource_props: {
+    'ensure'    => 'present',
+    'ssm_range' => '224.0.0.0/8 225.0.0.0/8',
+  },
+}
+
+tests['non_default_properties_S1'] = {
+  desc:           ' 1.1 Non default properties',
+  title_pattern:  'ipv4 red',
+  manifest_props: "
+      ssm_range   => '224.0.0.0/8',
+  ",
+  resource_props: {
+    'ensure'    => 'present',
+    'ssm_range' => '224.0.0.0/8',
+  },
+}
+
+tests['non_default_properties_S2'] = {
+  desc:           ' 1.2 Non default properties',
+  title_pattern:  'ipv4 black',
+  manifest_props: "
+      ssm_range   => '230.0.0.0/8',
+  ",
+  resource_props: {
+    'ensure'    => 'present',
+    'ssm_range' => '230.0.0.0/8',
+  },
+}
+
+tests['non_default_properties_S3'] = {
+  desc:           ' 1.3 Non default properties',
+  title_pattern:  'ipv4 default',
+  manifest_props: "
+      ssm_range   => 'none',
+  ",
+  resource_props: {
+    'ensure'    => 'present',
+    'ssm_range' => 'none',
+  },
+}
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+# Full command string for puppet resource with neighbor AF
+def puppet_resource_cmd(pim)
+  cmd = PUPPET_BINPATH + \
+        "resource cisco_pim '#{pim.values.join(' ')}'"
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_cisco_pim(tests, id)
+  manifest = prop_hash_to_manifest(tests[id][:pim])
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = { 'ensure' => 'absent' }
+  else
+    state = 'ensure => present,'
+    manifest += tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_cisco_pim :: title_pattern:\n" +
+               tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_pim { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_cisco_pim(tests, id)
+  pim = pim_title_pattern_munge(tests, id)
+  logger.info("\n--------\nTest Case Pim ID: #{pim}")
+
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd(pim)
+
+  # Build the manifest for this test
+  build_manifest_cisco_pim(tests, id)
+
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
```

Comment:
You can replace these 3 calls with a single call to `test_harness_common(tests, id)`.